deepchecks/vision/batch_wrapper.py

```diff
@@ -49,15 +50,28 @@ def labels(self):
             self._labels = dataset.batch_to_labels(self._batch)
         return self._labels
 
+    def _do_static_pred(self):
+        preds = self._context.static_predictions[self._dataset_kind]
+        dataset = self._context.get_data_by_kind(self._dataset_kind)
+        indexes = [dataset.to_dataset_index(self.batch_start_index + index)[0]
+                   for index in range(len(self))]
+        if isinstance(preds, torch.Tensor):
+            return preds[indexes]
+        return itemgetter(*indexes)(preds)
+
     @property
     def predictions(self):
         """Return predictions for the batch, formatted in deepchecks format."""
         if self._predictions is None:
             dataset = self._context.get_data_by_kind(self._dataset_kind)
-            # Calling model will raise error if model was not given
-            model = self._context.model
-            self._context.assert_predictions_valid(self._dataset_kind)
-            self._predictions = dataset.infer_on_batch(self._batch, model, self._context.device)
+            if self._context.static_predictions is not None:
+                self._context.assert_predictions_valid(self._dataset_kind)
```

Comment:
why do you have the same `self._context.assert_predictions_valid(self._dataset_kind)` twice?

Reply:
fixed comment bellow